hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/HazelcastClientCachingProvider.java

```diff
@@ -147,34 +161,34 @@ private HazelcastInstance getOrCreateByInstanceName(String instanceName) {
 
     private ClientConfig getDefaultClientConfig() {
         ClientConfig clientConfig = new XmlClientConfigBuilder().build();
-        if (namedDefaultHzInstance && StringUtil.isNullOrEmpty(clientConfig.getInstanceName())) {
+        if (namedDefaultHzInstance && isNullOrEmpty(clientConfig.getInstanceName())) {
             clientConfig.setInstanceName(SHARED_JCACHE_INSTANCE_NAME);
         }
         return clientConfig;
     }
 
-    protected ClientConfig getConfigFromLocation(String location, ClassLoader classLoader, String instanceName)
+    private ClientConfig getConfigFromLocation(String location, ClassLoader classLoader, String instanceName)
             throws URISyntaxException, IOException {
         URI uri = new URI(location);
         return getConfigFromLocation(uri, classLoader, instanceName);
     }
 
-    protected ClientConfig getConfigFromLocation(URI uri, ClassLoader classLoader, String instanceName)
+    private ClientConfig getConfigFromLocation(URI location, ClassLoader classLoader, String instanceName)
             throws URISyntaxException, IOException {
-        String scheme = uri.getScheme();
+        String scheme = location.getScheme();
         if (scheme == null) {
-            // it is a place holder
-            uri = new URI(System.getProperty(uri.getRawSchemeSpecificPart()));
-            scheme = uri.getScheme();
+            // interpret as place holder
+            location = new URI(System.getProperty(location.getRawSchemeSpecificPart()));
+            scheme = location.getScheme();
         }
         ClassLoader theClassLoader = classLoader == null ? getDefaultClassLoader() : classLoader;
         URL configURL;
         if ("classpath".equals(scheme)) {
-            configURL = theClassLoader.getResource(uri.getRawSchemeSpecificPart());
+            configURL = theClassLoader.getResource(location.getRawSchemeSpecificPart());
         } else if ("file".equals(scheme) || "http".equals(scheme) || "https".equals(scheme)) {
-            configURL = uri.toURL();
+            configURL = location.toURL();
         } else {
-            throw new URISyntaxException(uri.toString(), "Unsupported protocol in configuration location URL");
+            throw new URISyntaxException(location.toString(), "Unsupported protocol in configuration location URL");
         }
         try {
             return getConfig(configURL, classLoader, instanceName);
```

Comment:
By comparing the client and server implementation, I saw that we use `theClasserLoader` instead of `classLoader` on the server side, to get the config. Is this a bug in the client implementation?
```java
return getConfig(configURL, classLoader, instanceName)
```
vs.
```java
return getConfig(configURL, theClassLoader, instanceName)
```
If not, we might want to document why the `classLoader` should be used on the client side.

Reply:
Looks like it was introduced here: https://github.com/hazelcast/hazelcast/pull/9973/files#diff-0221acdabbaaa617b0fad8bd53fc584eR149

Reply:
Indeed it's a discrepancy vs server-side, but seems to have little or no effect at all: when `classLoader` is `null`, then `theClassLoader` is assigned  to the one that loaded hazelcast classes. A `null` classloader in `ClientConfig` essentially has the same effect - it ends up being used to resolve classes via `ClassLoaderUtil#newInstance` -- with a `null` classloader, it also uses the one that loaded Hazelcast classes for class resolution. Nevertheless, for uniformity's sake, let's have the client-side use `theClassLoader`.

Reply:
Good explanation. As you said, I'll change it to `theClassLoader`, so the classes are easier to compare. If I make a diff, the only difference now is `Config` vs. `ClientConfig`, naming (server/client) and which kind of `HazelcastInstance` is created (member/client).